extensions/amp-subscriptions/0.1/platform-store.js

```diff
@@ -242,6 +242,58 @@ export class PlatformStore {
     return this.entitlementDeferredMap_[serviceId].promise;
   }
 
+  /**
+   * Get scoreFactor states for each platform
+   * @return {Promise<!JsonObject>}
+   *
+   * return value looks somethinglike this
+   * {
+   *    isReadyToPay: {
+   *     'subscription-google-com': 1,
+   *     local: 0,
+   *    },
+   *    supportdViewer: {
+   *     'subscription-google-com': 1,
+   *     local: 0,
+   *    },
+   * }
+   */
+  getScoreFactorStates() {
+    const states = dict({});
+    return Promise.all(
+      Object.values(SubscriptionsScoreFactor).map(scoreFactor => {
+        states[scoreFactor] = dict();
+        return Promise.all(
+          this.serviceIds_.map(platformId =>
+            this.getScoreFactorPromiseFor_(platformId, scoreFactor).then(
+              factorValue => {
+                states[scoreFactor][
+                  platformId.replace(/\./g, '_')
```

Comment:
Dash or underscore? Above it documents "dash".

Reply:
underscore, the expression parser throws on dash

Reply:
I could also support "dash" in names if that helps. There's a lot of precedent for dashed names in HTML/XML and family.

Reply:
Dash would be good too, although supporting the `['name.with.dots']` format will remove that need